src/main/scala/mesosphere/marathon/core/appinfo/TaskCounts.scala

```diff
@@ -13,16 +13,34 @@ import org.apache.mesos.{ Protos => mesos }
   * @param tasksUnhealthy snapshot of the number of unhealthy tasks (does not include tasks without health info)
   */
 case class TaskCounts(
-  tasksStaged: Int,
-  tasksRunning: Int,
-  tasksHealthy: Int,
-  tasksUnhealthy: Int)
+    tasksStaged: Int,
+    tasksRunning: Int,
+    tasksHealthy: Int,
+    tasksUnhealthy: Int) {
+  def +(counts: TaskCounts): TaskCounts = {
+    copy(
+      tasksRunning = tasksRunning + counts.tasksRunning,
+      tasksStaged = tasksStaged + counts.tasksStaged,
+      tasksHealthy = tasksHealthy + counts.tasksHealthy,
+      tasksUnhealthy = tasksUnhealthy + counts.tasksUnhealthy
+    )
+  }
+
+  def -(counts: TaskCounts): TaskCounts = {
+    copy(
+      tasksRunning = tasksRunning - counts.tasksRunning,
+      tasksStaged = tasksStaged - counts.tasksStaged,
+      tasksHealthy = tasksHealthy - counts.tasksHealthy,
+      tasksUnhealthy = tasksUnhealthy - counts.tasksUnhealthy
```

Comment:
For the sake of the subtraction, the implementation is fine – semantically, something like `TaskCounts.zero - TaskCounts(1, 1, 1, 1)` should at least result in `TaskCounts.zero`, not in `TaskCounts(-1, -1, -1, -1)`